.github/workflows/pr.yml

```diff
@@ -1,9 +1,15 @@
-name: PR Build
+name: "Build and Test"
 on:
   pull_request:
     branches:
       - master
       - staging
+  push:
+    branches:
+      - master
+      - staging
+env:
+  CROSS_BUILD: ${{ github.ref == 'refs/heads/staging' || github.ref == 'refs/heads/master' }}
```

Comment:
As I understand: we want do cross build only when we merge our branch to master or staging?

Reply:
I think there should be also checking preview branch like below - maybe we can move this piece of code to separate fragment?